pkg/plugins/runtime/k8s/controllers/gateway_instance_controller.go

```diff
@@ -92,14 +92,14 @@ func (r *GatewayInstanceReconciler) createOrUpdateService(
 		return selector.Matches(gatewayInstance.Spec.Tags)
 	})
 
-	if gateway == nil {
-		// We have an index and watch set up to requeue when this changes
-		return nil, nil
-	}
-
 	obj, err := ctrls_util.CreateOrUpdateControlled(
 		ctx, r.Client, gatewayInstance, &kube_core.ServiceList{},
 		func(obj kube_client.Object) (kube_client.Object, error) {
+			// If we don't have a gateway, we don't want our Service anymore
+			if gateway == nil {
+				return nil, nil
+			}
```

Comment:
WDYT about adding a `ctrls_util.DeleteControlled()` - would that make the behavior more obvious?

Reply:
I added more to the godoc and renamed the function. I personally like the abstraction of having `mutate` decide whether to delete the object by explicitly returning `nil` as an option type, vs having to call multiple, more "imperative" functions.

Reply:
OK, I think the renaming makes sense :+1: